File: Flutter/App/lib/data/product/mapper/data_to_domain_product_details_mapper.dart
import 'package:products_crud/data/product/model/product_details_dto.dart';
import 'package:products_crud/domain/product/model/product_details.dart';

extension ProductDetailsMapper on ProductDetailsDTO {
  ProductDetails toDomain() {
    return ProductDetails(
      id: id,
      title: title,
      description: description,
      price: price,
      category: category,
      images: images,
    );
  }
}

File: Android/App/app/src/main/java/com/efalco777/crud_products/data/product/mapper/DataToDomainProductDetailsMapper.kt
package com.efalco777.crud_products.data.product.mapper

import com.efalco777.crud_products.data.product.model.ProductDTO
import com.efalco777.crud_products.data.product.model.ProductDetailsDTO
import com.efalco777.crud_products.domain.product.model.Product
import com.efalco777.crud_products.domain.product.model.ProductDetails

class DataToDomainProductDetailsMapper {
    companion object {
        fun map(model: ProductDetailsDTO): ProductDetails {
            return ProductDetails(
                id = model.id,
                title = model.title,
                description = model.description,
                category = model.category,
                price = model.price
            )
        }
    }
}